### PR TITLE
Provide a mechanism to indicate that a change is considered obvious

### LINKED ISSUE #132: Provide a mechanism to indicate that a change is considered obvious
It would be handy of a project collaborator could indicate that a change is considered obvious and that the CLA does not need to be signed. Perhaps via a comment directed at `pivotal-issuemaster`:

```
@pivotal-issuemaster This is an obvious fix
```

The CLA bot could then update the PR to remove the failing CLA check and perhaps make a follow-up comment that the CLA doesn't need to be signed after all.

### PR DESCRIPTION
**Pull request by mp911de**
Users can now add a comment (issue comment/review comment within a commit) that the change contains an "obvious fix". We no longer require the CLA to be signed. In case the CLA is signed we keep the regular flow and update the status up CLA signed. If the user is a first-time user and the PR has a comment saying "Please sign..." we add a disarming comment stating "CLA signing not necessary" and update the commit status to "CLA not required".

Fixes gh-132

### CODE REVIEW

**Comment by Shredder121** [comment]
I think we're still missing the "obvious fix" text that could be present in the PR body (`PullRequestEvent` `opened`/`edited`).
Or am I not looking carefully enough?

**Comment by mp911de** [comment]
Good catch. I added another commit to catch `obvious fix` in the PR body.

**Comment by Shredder121** [comment]
And what about the PR title?

I'm sure there are people who would name their PRs something like `[Obvious fix] flip null check`.

**Comment by Shredder121** [comment]
As a side note, are lombok's field defaults not working for you?
Just wondering.

**Comment by mp911de** [comment]
I checked some PR's inside various repositories. There are no PR's named with that pattern. I think we should not allow too much possibilities up-front. 

Not sure I follow regarding

> are lombok's field defaults not working for you?

**Comment by Shredder121** [comment]
> I think we should not allow too much possibilities up-front.

Fine by me, you're right.

> Not sure I follow regarding
> 
> > are lombok's field defaults not working for you?

As of #104 lombok helps making fields private by default, making private declarations not necessary anymore.
I saw you added the access modifiers in for example [PullRequestId](https://github.com/pivotalsoftware/pivotal-cla/pull/135/files#diff-07b534a039bec1aa03c519264c15f7d7R28), which is okay, a matter of taste I suppose, although `@Value` already makes every field `private` and `final`.
But, I also noticed [that you put back a few `private` keywords](https://github.com/pivotalsoftware/pivotal-cla/pull/135/files#diff-4d0a7028977cebcede41d5af9a72652aR91) which lead me to ask that question.

**Comment by mp911de** [comment]
Got it. That's just a matter of not paying enough attention to the recent changes. Thanks for the hint.

File: src/main/java/io/pivotal/cla/service/github/MylynGitHubApi.java [hunk 1]
```diff
@@ -65,18 +69,28 @@
...
+	public final static String OBVIOUS_FIX_CLA_NOT_REQUIRED = String.format(
+			"This Pull Request contains an obvious fix. Signing the %s not necessary.", CONTRIBUTOR_LICENSE_AGREEMENT);
+	public final static String THANK_YOU = "Thank you for signing the";
+	public final static String PLEASE_SIGN = "Please sign the";
+	public final static String OBVIOUS_FIX = "obvious fix";
 
-	RestTemplate rest = new RestTemplate();
+	private ClaOAuthConfig oauthConfig;
+	private String authorizeUrl;
+	private  RestTemplate rest = new RestTemplate();
```
Comment (Shredder121) [inline src/main/java/io/pivotal/cla/service/github/MylynGitHubApi.java]: And what about these?

Comment (Shredder121) [inline src/main/java/io/pivotal/cla/service/github/MylynGitHubApi.java]: Sorry if I am a burden.

Comment (mp911de) [inline src/main/java/io/pivotal/cla/service/github/MylynGitHubApi.java]: No worries. Always happy if someone takes another look. That component is not a `@Value` object and it shouldn't be `@Value`. The general idea of a value object (immutability, limit visibility) would apply here but the value idea is misleading.

Comment (Shredder121) [inline src/main/java/io/pivotal/cla/service/github/MylynGitHubApi.java]: I meant, these `private` modifiers were put back.

Comment (mp911de) [inline src/main/java/io/pivotal/cla/service/github/MylynGitHubApi.java]: There's no reason why the fields of `MylynGitHubApi` should be visible to other classes.

Comment (Shredder121) [inline src/main/java/io/pivotal/cla/service/github/MylynGitHubApi.java]: But Lombok makes them private by default.

**Comment by mp911de** [comment]
The IntelliJ plugin has an issue with the `lombok.config`. Errors get displayed only after compiling. D'oh!

**Comment by Shredder121** [comment]
> The IntelliJ plugin has an issue with the lombok.config.

Yeahh, that's possible.

Sorry for the nagging though.

**Comment by mp911de** [comment]
I found and fixed a bug that was located in filtering "own" events. The check now considers the `sender` field to check the user whether to skip the event.

**Comment by Shredder121** [comment]
> The check now considers the sender field

Yes sounds good.
Although, what do you do if the `pull_request` event is sent by a bot, instead of the user?
Then would the bot need to submit a CLA? Or would you also take the git `author` in account?

**Comment by Shredder121** [comment]
Hmm, nevermind, didn't look good enough.
It's a little late, my bad.

**Comment by mp911de** [comment]
The current thinking is: The creator of the PR (= the user who opens the PR) should have signed the PR. There are however other aspects to that: What about the committing user? What if there are multiple commits from different people?

As long as we don't consider the commits we just care whether the user, who opened the PR, has signed the CLA - no matter whether it's a bot or not.

**Comment by rwinch** [comment]
Thanks for the PR! This is merged via 1ff2eff7f695c0cb9044aff4fd194d6718a4021a

**Pull request closed without merging**